app/Http/Controllers/AgreementController.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Auth;
+use App\Agreement;
+
+class AgreementController extends Controller
+{
+
+    private function get_agreement(){
+        $tos = Agreement::where("in_effect_at", "<=", now())->latest()->first();
```

Comment:
Inclined to think the logic for "how we get latest agreement in effect" should be a method on `Agreement`. Something like `Agreement->current()` that would just get the most current agreement that is in effect (since any previous ones aren't any longer).